### ISSUE TITLE
Proposal: Warn about using commas with printf

### ISSUE THREAD

**Issue by kmmiles** [koalaman/shellcheck#2213]
I make this mistake constantly. Is it possible for `shellcheck` to warn about it?

```sh
#!/bin/bash
printf "%s\n", "$PATH"
```

**Comment by koalaman** [comment]
I really like this idea. I ran some experiments though, and it doesn't appear to be a systemic issue. The majority of shell words with a trailing, unquoted comma are things like `cut -d,`  `tr , ' '`, `echo Foo, bar, etc`. When only looking at quoted text + unquoted literal `,` the accuracy better, but true positives were mostly things like `foo=("bar", "baz")` already covered by SC2054, plus several false positives like `echo "$foo", "$bar", etc`. I was not able to find any differentiator that resulted in more true than false positives.

Thanks for the suggestion!
